R/run-nmtran.R

```diff
@@ -0,0 +1,330 @@
+#' Interface for running `NM-TRAN` on model objects
+#'
+#' Function to run `NM-TRAN` on a model object to validate its control stream
+#'  for correct coding before submission. The `NM-TRAN` dataset (`FDATA`) and
+#'  other `NONMEM` artifacts can be further inspected by keeping the run directory
+#'  around.
+#'
+#' @details
+#' `NM-TRAN` is a preprocessor for `NONMEM` that translates user-specified
+#' control stream data and instructions into a form executable by `NONMEM`.
+#'
+#' Note that `nmtran_presort` is run ahead of `NM-TRAN` for `NONMEM` versions
+#' `'nm74gf'`, `'nm74gf_nmfe'`, and `'nm75'`.
+#'  - `nmtran_presort` is an supplementary utility that preprocesses the control
+#'     stream to ensure it is in the correct format for `NM-TRAN`. It is particularly
+#'     relevant for handling specific data manipulations and ensuring compatibility
+#'     with the `NM-TRAN` executable.
+#'
+#' @param .mod A `bbi_nonmem_model` object.
+#' @param .bbi_args A named list specifying arguments to pass to `NM-TRAN`.
+#'   Similar to the `.bbi_args` argument defined in [submit_model()], though here
+#'   only `prdefault`, `tprdefault`, and `maxlim` flags are passed to `NM-TRAN`.
+#'   See [print_bbi_args()] for more details.
+#' @inheritParams submit_model
+#' @param .nonmem_version Character scalar for default version of NONMEM to use.
+#'   If left `NULL`, will look for the default version specified in the provided
+#'   `bbi` configuration file.
+#' @param clean Logical (`T`/`F`). If `FALSE`, don't delete the temporary folder
```

Comment:
This is minor and I know our docs say "folder" elsewhere, but I'd use "directory" to lean toward what I think is much more idiomatic for R and matches the `*nix` terminology.

Some support for this:

```
### in core R source
$ git grep '\bdirectory\b' '*.R'  | wc -l
     301

$ git grep '\bfolder\b' '*.R'  | wc -l
       3

# all Windows-specific contexts
$ git grep '\bfolder\b' '*.R'
src/gnuwin32/installer/WiXins.R:## The standard folder names are listed at
src/library/tools/R/QC.R:  , choose.dir = function(default = "", caption = "Select folder") {}
src/library/utils/R/windows/choose.files.R:choose.dir <- function(default = '', caption = 'Select folder')

### in fs repo
$ git grep '\bdirectory\b' '*.R' | wc -l
      75
$ git grep '\bfolder\b' '*.R' | wc -l
       3

$ git grep '\bfolder\b' '*.R'
tests/testthat/test-fs_path.R:      # folder
tests/testthat/test-path.R:    x <- "folder//fa\xE7ile.txt/"
tests/testthat/test-path.R:    expect_equal(out, fs_path("folder/façile.txt"))
```